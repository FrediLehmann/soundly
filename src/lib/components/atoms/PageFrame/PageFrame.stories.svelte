<script>
  import { Meta, Story, Template } from '@storybook/addon-svelte-csf';

  import PageFrame from './PageFrame.svelte';
</script>

<Meta
  title="Atoms/PageFrame"
  component={PageFrame}
  argTypes={{
    style: {
      control: { type: 'select' },
      options: ['default', 'small'],
      table: {
        defaultValue: {
          summary: 'default'
        },
        type: {
          summary: 'The style of the PageFrame'
        }
      }
    }
  }} />

<Template let:args>
  <PageFrame {...args}><div class="content">Content</div></PageFrame>
</Template>

<Story name="Default" args={{ style: 'default' }} />

<Story name="Small" args={{ style: 'small' }} />

<style>
  .content {
    background-color: green;
    width: 100%;
    height: 100%;
    padding: 12px;
  }
</style>
