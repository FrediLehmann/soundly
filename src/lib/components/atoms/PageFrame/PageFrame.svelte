<script lang="ts">
  import { PageFrameTypes } from './PageFrameTypes.enum';

  export let type: PageFrameTypes = PageFrameTypes.default;
</script>

<main class={`${$$props.class ? `${$$props.class} ` : ''}${type}`}>
  <slot />
</main>
