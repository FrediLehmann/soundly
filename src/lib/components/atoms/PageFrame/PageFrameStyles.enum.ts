export enum PageFrameStyles {
  default = 'max-w-4xl mx-auto pt-4 px-4 sm:px-8 lg:px-10',
  small = 'max-w-lg mx-auto px-4 sm:px-8 lg:px-10'
}
