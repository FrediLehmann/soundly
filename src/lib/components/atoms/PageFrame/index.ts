import PageFrame from './PageFrame.svelte';

export { PageFrame };
