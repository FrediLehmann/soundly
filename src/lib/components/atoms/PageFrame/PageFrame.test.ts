/**
 * @jest-environment jsdom
 */

import '@testing-library/jest-dom';
import { render } from '@testing-library/svelte';

import PageFrame from './PageFrame.svelte';
import { PageFrameStyles } from './PageFrameStyles.enum';

describe('PageFrame component', () => {
  test('redners default PageFrame', () => {
    const { getByRole } = render(PageFrame);
    expect(getByRole('main')).toBeInTheDocument();
    expect(getByRole('main').className).toEqual(PageFrameStyles.default);
  });

  test('render correct styles', () => {
    for (const style in PageFrameStyles) {
      const rendered = render(PageFrame, {
        style
      });

      expect(rendered.getByRole('main').className).toEqual(style);
      rendered.unmount();
    }
  });

  test('custom class', () => {
    const { getByRole } = render(PageFrame, { class: 'added-class' });
    expect(getByRole('main').className).toEqual(
      `added-class ${PageFrameStyles.default}`
    );
  });
});
