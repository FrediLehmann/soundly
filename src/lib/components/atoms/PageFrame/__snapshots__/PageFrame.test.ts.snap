// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageFrame /> component redners PageFrame 1`] = `
PageFrame {
  "$$": Object {
    "after_update": Array [],
    "before_update": Array [],
    "bound": Object {},
    "callbacks": Object {},
    "context": Map {},
    "ctx": Array [
      "default",
      Object {},
      undefined,
      Object {},
    ],
    "dirty": Array [
      -1,
    ],
    "fragment": Object {
      "c": [Function],
      "d": [Function],
      "i": [Function],
      "l": [Function],
      "m": [Function],
      "o": [Function],
      "p": [Function],
    },
    "not_equal": [Function],
    "on_destroy": Array [
      [Function],
    ],
    "on_disconnect": Array [],
    "on_mount": Array [],
    "props": Object {
      "style": 0,
    },
    "root": <div>
      <main
        class="svelte-vhl70c"
        data-style="default"
      />
    </div>,
    "skip_bound": false,
    "update": [Function],
  },
  "$$set": [Function],
  "$capture_state": [Function],
  "$inject_state": [Function],
}
`;
